tests/functional/suite_test.go

```diff
@@ -120,13 +127,27 @@ var _ = BeforeSuite(func() {
 	th = NewTestHelper(ctx, k8sClient, timeout, interval, logger)
 	Expect(th).NotTo(BeNil())
 
+	// Start the controller-manager if goroutine
+	webhookInstallOptions := &testEnv.WebhookInstallOptions
 	k8sManager, err := ctrl.NewManager(cfg, ctrl.Options{
 		Scheme: scheme.Scheme,
+		// NOTE(gibi): disable metrics reporting in test to allow
+		// parallel test execution. Otherwise each instance would like to
+		// bind to the same port
+		MetricsBindAddress: "0",
+		Host:               webhookInstallOptions.LocalServingHost,
+		Port:               webhookInstallOptions.LocalServingPort,
+		CertDir:            webhookInstallOptions.LocalServingCertDir,
+		LeaderElection:     false,
 	})
 	Expect(err).ToNot(HaveOccurred())
 
 	kclient, err := kubernetes.NewForConfig(cfg)
 	Expect(err).ToNot(HaveOccurred(), "failed to create kclient")
+
+	err = (&heat.Heat{}).SetupWebhookWithManager(k8sManager)
+	Expect(err).NotTo(HaveOccurred())
```

Comment:
I know this is testing a validation webhook instead of a defaulting one, but since we're adding the webhook server to the functional test, should we perhaps fully initialize everything required for the webhooks to function properly (in case someone later adds tests for the defaulting webhooks)?  I'm thinking specifically of this [1].

[1] https://github.com/openstack-k8s-operators/heat-operator/blob/a7985c1c308b3fed34dd36358d55864f978d5be2/main.go#L159-L160

Reply:
That makes sense. Because this work is now pending, I've created PR to add tests for the defaulting webhooks.

https://github.com/openstack-k8s-operators/heat-operator/pull/179